### PR TITLE
Fix bug where `PLE1307` was raised when formatting `%c` with characters

### PR DESCRIPTION
**Pull request by zanieb**
Closes https://github.com/astral-sh/ruff/issues/8406

### CODE REVIEW

File: crates/ruff_linter/resources/test/fixtures/pylint/bad_string_format_type.py [hunk 1]
```diff
@@ -26,9 +26,7 @@
 "%d %s %f" % VALUES_TO_FORMAT
 "%s" % "1"
 "%s %s %s" % ("1", 2, 3.5)
-print("%d %d"
-      %
-(1, 1.1))
+print("%d %d" % (1, 1.1))
```
Comment (zanieb) [inline crates/ruff_linter/resources/test/fixtures/pylint/bad_string_format_type.py:29]: Okay Ruff formatter :D will revert

File: crates/ruff_linter/src/rules/pylint/rules/bad_string_format_type.rs [hunk 2]
```diff
@@ -119,12 +119,23 @@ fn collect_specs(formats: &[CFormatStrOrBytes<String>]) -> Vec<&CFormatSpec> {
...
-    let format = FormatType::from(format.format_char);
+    let format_type = FormatType::from(format.format_char);
     match ResolvedPythonType::from(value) {
-        ResolvedPythonType::Atom(atom) => format.is_compatible_with(atom),
-        ResolvedPythonType::Union(atoms) => {
-            atoms.iter().all(|atom| format.is_compatible_with(*atom))
+        ResolvedPythonType::Atom(atom) => {
+            // Special case where `%c` allows single character strings to be formatted
+            if format.format_char == 'c' {
+                if let Expr::StringLiteral(string) = value {
```
Comment (MichaReiser) [inline crates/ruff_linter/src/rules/pylint/rules/bad_string_format_type.rs:127]: Could this also be a byte string or fstring?

**Review by MichaReiser — APPROVED** [review]

File: crates/ruff_linter/resources/test/fixtures/pylint/bad_string_format_type.py [hunk 3]
```diff
@@ -57,3 +57,9 @@
 '(%r, %r, %r, %r)' % (hostname, address, username, '$PASSWORD')
 '%r' % ({'server_school_roles': server_school_roles, 'is_school_multiserver_domain': is_school_multiserver_domain}, )
 "%d" % (1 if x > 0 else 2)
+
+# Special cases for %c allowing single character strings
+# https://github.com/astral-sh/ruff/issues/8406
+"%c" % ("x",)
+"%c" % "x"
+"%c" & "œ"
```
Comment (dhruvmanila) [inline crates/ruff_linter/resources/test/fixtures/pylint/bad_string_format_type.py:65]: Probably a typo?
```suggestion
"%c" % "œ"
```

**Review by dhruvmanila — APPROVED** [review]

File: crates/ruff_linter/resources/test/fixtures/pylint/bad_string_format_type.py (hunk 3 above)
Comment (zanieb) [reply crates/ruff_linter/resources/test/fixtures/pylint/bad_string_format_type.py]: Definitely, thanks!

**Pull request merged**